src/rgw/rgw_sync_module_es.cc

```diff
@@ -801,6 +817,25 @@ class RGWElasticDataSyncModule : public RGWDataSyncModule {
 
   void init(RGWDataSyncEnv *sync_env, uint64_t instance_id) override {
     conf->init_instance(sync_env->store->svc.zone->get_realm(), instance_id);
+    // try to get elastic search version
```

Comment:
is this block necessary, can't we just initialize conf->es_info in RGWElasticInitConfigCBCR itself?

Reply:
Yes, It's necessary, `RGWElasticInitConfig` is invoked at `data sync init` once.

Reply:
Since we already get the version info at `RGWElasticInitConfigCBCR` which is initiated once at the start of RGW, can't we just copy the es_info struct we have there on to the conf struct. instead of the need to parse the ES version twice?

Reply:
We could not get `es_info` when data sync is already in `Inc Sync` status.  RGW will not re-init data sync when the process is restarted.

Reply:
What happens when you restart the GW? wont InitConfig be invoked again? Or is it really only once per install?

Reply:
yes, It is really only once per install. 

see https://github.com/ceph/ceph/blob/cc17681b478594aa39dd80437256a54e388432f0/src/rgw/rgw_data_sync.cc#L1630-L1634

`RGWElasticInitConfig` is called only by `init_sync` function.  we invoke `init_sync` func if and only if sync status is `StateBuildingFullSyncMaps`